generator/generator_tests/metadata_parser_test.cpp

```diff
@@ -0,0 +1,199 @@
+#include "testing/testing.hpp"
+#include "base/logging.hpp"
+
+#include "indexer/classificator_loader.hpp"
+#include "generator/osm2meta.hpp"
+#include "coding/writer.hpp"
+#include "coding/reader.hpp"
+
+#include "std/target_os.hpp"
+
+using feature::Metadata;
+
+UNIT_TEST(Metadata_ValidateAndFormat_stars)
+{
+  FeatureParams params;
+  MetadataTagProcessor p(params);
+  Metadata & md = params.GetMetadata();
+
+  // ignore incorrect values
```

Comment:
This code is not mine, I've moved it here from another place without changes.

But I'll fix it anyway.